### DISCUSSION TITLE
Android Play Store build - Expo

### DISCUSSION THREAD

**Discussion by paulinha-19** [expo/expo#25285] [Archived: Questions]
I need to build my application expo maneged using eas. In the vs code terminal, enter the commands below:
1) npm install -g eas-cli
2) eas login
3) eas build:configure
  √ Which platforms would you like to configure for EAS Build? » Android
√ Generated eas.json. Learn more: https://docs.expo.dev/build-reference/eas-json/
🎉 Your project is ready to build.
- Run eas build when you are ready to create your first build.
- Once the build is completed, run eas submit to upload the app to app stores.
- Learn more about other capabilities of EAS Build: https://docs.expo.dev/build/introduction
4) eas build --platform android
 Loaded "env" configuration for the "production" profile: no environment variables specified. Learn more: https://docs.expo.dev/build-reference/variables/
✔ Using remote Android credentials (Expo server)
✔ Using Keystore from configuration: Build Credentials wdhdOKG1L- (default)
Compressing project files and uploading to EAS Build. Learn more: https://expo.fyi/eas-build-archive
✔ Uploaded to EAS 1s
Waiting for build to complete. You can press Ctrl+C to exit.
✖ Build failed
🤖 Android build failed:
Gradle build failed with unknown error.

On the expo website I received the errors below.

**Run gradlew**
```
Running 'gradlew :app:bundleRelease' in /home/expo/workingdir/build/android

Downloading https://services.gradle.org/distributions/gradle-8.0.1-all.zip

10%

20%

30

%.

40%

50%.

60%

70%.

80%

90%

100%

Welcome to Gradle 8.0.1!

Here are the highlights of this release:

 - Improvements to the Kotlin DSL

- Fine-grained parallelism from the first build with configuration cache

- Configurable Gradle user home cache cleanup

For more details see https://docs.gradle.org/8.0.1/release-notes.html

To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/8.0.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.

Daemon will be stopped at the end of the build

> Task :gradle-plugin:pluginDescriptors

> Task :gradle-plugin:processResources

> Task :gradle-plugin:compileKotlin

> Task :gradle-plugin:compileJava NO-SOURCE

> Task :gradle-plugin:classes

> Task :gradle-plugin:jar

> Task :gradle-plugin:inspectClassesForKotlinIC

FAILURE: Build completed with 2 failures.

1:

Task failed with an exception.

-----------

* Where:

Build file '/home/expo/workingdir/build/node_modules/expo-error-recovery/android/build.gradle' line: 40

* What went wrong:

A problem occurred evaluating project ':expo-error-recovery'.

>

Could not set unknown property 'classifier' for task ':expo-error-recovery:androidSourcesJar' of type org.gradle.api.tasks.bundling.Jar.

* Try:

>

Run with --stacktrace option to get the stack trace.

> Run with

--info

or --debug option to get more log output.

>

Run with

--scan to get full insights.

==============================================================================

2: Task failed with an exception.

-----------

* What went wrong:

A problem occurred configuring project ':expo'.

> compileSdkVersion is not specified. Please add it to build.gradle

* Try:

> Run with --stacktrace option to get the stack trace.

>

Run with

--info or --debug option to get more log output.

> Run with

--scan

to get full insights.

==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 54s

5 actionable tasks: 5 executed

Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

**Fail build**
```
Build failed: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

**Comment by swayok** [comment]
I have exactly the same problem and I cannot find out what causes this. My guesses are: sentry-expo or expo-updates. I have reviewed change logs and updated everything related to new SDK version but it still fails. Will try to remove sentry and updates, maybe this will help to find out where is the problem...

**Comment by swayok** [comment]
Just rolled back to Expo 47. It uses Gradle 7.5.1 and build was successful. So there seems to be some problem with upgrading. I have reviewed changes though and have no idea what may be the reason.

**Comment by swayok** [comment]
@paulinha-19 I have found the reason of this error:
Package `expo-error-recovery` now deprecated. For some reason I had it in my `package.json` since a long time ago. So I removed it and build was successful.
If you do not have this package - use this Linux command to find which package requires it:
`find ./node_modules/ -name package.json | xargs grep expo-error-recovery`
Now you will know what package is deprecated and decide what to do with it.
